src/util/Algorithm.h

```diff
@@ -80,6 +81,29 @@ auto transform(Range&& input, F unaryOp) {
   return out;
 }
 
+/*
+@brief Takes two vectors, pairs up their content at the same index positions
+and copies them into `std::pair`s, who are returned inside a vector.
+Example: `{1,2}` and `{3,4}` are returned as `{(1,3), (2,4)}`.
+*/
+template <typename VectorAContentType, typename VectorBContentType>
+std::vector<std::pair<VectorAContentType, VectorBContentType>>
+pairTwoVectorTogether(const std::vector<VectorAContentType>& vectorA,
```

Comment:
A shorter and better name would be `zipVectors(...)`

And you can simply use `T1, T2` instead of `VectorAContentType` and `...B..`

Reply:
Done, see [1e45e2a](https://github.com/ad-freiburg/qlever/pull/958/commits/1e45e2a2f428700686f4495f58d1b3c6793e3e98).